Fix broken docker/Dockerfile again

It seems that the fix of #38049 didn't work as expected see [this run for beta7](https://github.com/sagemath/sage/actions/runs/9234698759/job/25411312307#step:8:15396) and [this one for beta8](https://github.com/sagemath/sage/actions/runs/9333677050/job/25693858666#step:8:15441), explicitly in the first case:

```
#16 64.41 [sagemath_doc_html-none] [spkg-install] sage -advanced > en/reference/repl/options.txt
#16 64.41 [sagemath_doc_html-none] [spkg-install] Traceback (most recent call last):
#16 64.41 [sagemath_doc_html-none] [spkg-install]   File "<frozen runpy>", line 198, in _run_module_as_main
#16 64.41 [sagemath_doc_html-none] [spkg-install]   File "<frozen runpy>", line 88, in _run_code
#16 64.41 [sagemath_doc_html-none] [spkg-install]   File "/home/sage/sage/local/var/lib/sage/venv-python3.11.8/lib/python3.11/site-packages/sage_docbuild/__main__.py", line 66, in <module>
#16 64.41 [sagemath_doc_html-none] [spkg-install]     from sage.env import SAGE_DOC_SRC
#16 64.41 [sagemath_doc_html-none] [spkg-install] ModuleNotFoundError: No module named 'sage'
#16 64.41 [sagemath_doc_html-none] [spkg-install] Error: 'sage --docbuild --all-documents' failed
#16 64.41 [sagemath_doc_html-none] [spkg-install] make[5]: *** [Makefile:41: doc-inventory-reference] Error 1
```

In the [run for beta4](https://github.com/sagemath/sage/actions/runs/8860984377/job/24333924316#step:8:5751) the module still was found.

@mkoeppe, do you have any idea what is missing?

_Originally posted by @soehms in https://github.com/sagemath/sage/issues/38049#issuecomment-2150516269_
            
This will be fixed as soon as #38304 is merged (see https://github.com/sagemath/sage/issues/38253#issuecomment-2203791462),

--- a/build/pkgs/sagelib/spkg-install.in
+++ b/build/pkgs/sagelib/spkg-install.in
@@ -26,9 +26,9 @@ export SAGE_SRC_ROOT=/doesnotexist
 export SAGE_DOC_SRC=/doesnotexist
 export SAGE_BUILD_DIR=/doesnotexist
 
-# We also poison all directories below SAGE_LOCAL.
+# We also poison all directories below SAGE_LOCAL
+# except for SAGE_SPKG_SCRIPTS, which is needed by sage-dist-helpers
 export SAGE_PKGCONFIG=/doesnotexist
-export SAGE_SPKG_SCRIPTS=/doesnotexist
 export SAGE_SHARE=/doesnotexist
 
 # However, we only unset SAGE_LOCAL and SAGE_PKG_CONFIG_PATH.

--- a/build/pkgs/sagelib/spkg-install.in
+++ b/build/pkgs/sagelib/spkg-install.in
@@ -24,7 +24,6 @@ export SAGE_ROOT=/doesnotexist
 export SAGE_SRC=/doesnotexist
 export SAGE_SRC_ROOT=/doesnotexist
 export SAGE_DOC_SRC=/doesnotexist
-export SAGE_BUILD_DIR=/doesnotexist
 
 # We also poison all directories below SAGE_LOCAL
 # except for SAGE_SPKG_SCRIPTS, which is needed by sage-dist-helpers

